protostar/cli/signable_command_util.py

```diff
@@ -1,97 +1,80 @@
+        module = ".".join(module_names)
+        signer_module = importlib.import_module(module)
+        signer_class = getattr(signer_module, class_name)
+        _validate_signer_interface(signer_class)
+        return signer_class()
 
-class SignableCommandUtil:
-    signable_arguments = [
-        ProtostarArgument(
-            name="account-address",
-            description="Account address.",
-            type="address",
-            is_required=True,
-        ),
-        ProtostarArgument(
-            name="private-key-path",
-            description="Path to the file, which stores your private key (in hex representation) for the account. \n"
-            f"Can be used instead of {PRIVATE_KEY_ENV_VAR_NAME} env variable.",
-            type="path",
-        ),
-        ProtostarArgument(
-            name="signer-class",
-            description="Custom signer class module path.",
-            type="str",
-        ),
-    ]
+    private_key_str = None
+    if _args.private_key_path:
+        with open(_args.private_key_path, encoding="utf-8") as file:
+            private_key_str = file.read()
 
-    def __init__(self, args: Any):
-        self._args = args
+    if not private_key_str:
+        private_key_str = os.environ.get(PRIVATE_KEY_ENV_VAR_NAME)
 
-    def get_signer(
-        self,
-        network_config: NetworkConfig,
-    ) -> BaseSigner:
-        if self._args.signer_class:
-            *module_names, class_name = self._args.signer_class.split(".")
-            module = ".".join(module_names)
-            signer_module = importlib.import_module(module)
-            signer_class = getattr(signer_module, class_name)
-            SignableCommandUtil._validate_signer_interface(signer_class)
-            return signer_class()
+    if not private_key_str:
+        raise ProtostarException(
+            "Signing is mandatory, please provide a private key in order to sign transactions."
+        )
 
-        private_key_str = None
-        if self._args.private_key_path:
-            with open(self._args.private_key_path, encoding="utf-8") as file:
-                private_key_str = file.read()
+    try:
+        private_key_str = private_key_str.strip()
+        if not private_key_str.startswith("0x"):
+            private_key_str = hex(int(private_key_str))
+        private_key = int(private_key_str, 16)
+    except ValueError as v_err:
+        raise ProtostarException(
+            f"Invalid private key format ({private_key_str}). Please provide hex-encoded number."
+        ) from v_err
 
-        if not private_key_str:
-            private_key_str = os.environ.get(PRIVATE_KEY_ENV_VAR_NAME)
+    key_pair = KeyPair.from_private_key(private_key)
 
-        if not private_key_str or not self._args.account_address:
-            raise ProtostarException(
-                "Signing is mandatory, please provide an account and a private key in order to sign transactions."
-            )
+    account_address = (
```

Comment:
I think the logic here becomes a little convoluted, you can provide account address with two arguments. We could extract address from args in commands that have it and make `account_address` a required argument of `get_signer`. It would separate logic a little bit, as the way account-address is provided is command specific. What do you think?

Reply:
Agree, modified the function